File: 100-Days-Series/Day 48/LinkedListDS.js

class LinkedList{
    constructor(data){
        this.data=data;
        this.next=null;
    }
}

class LinkedListDS{
    constructor(){
        this.head=null;
    }
    addNode(data){
        const  newNode=new LinkedList(data);
        if(this.head==null){
            this.head=newNode;
            newNode.next=null;
        }else{
            let last;
            for(let temp=this.head;temp!=null;temp=temp.next){
                last=temp;
            }
            last.next=newNode;
            newNode.next=null;
        }
        console.log("New Node Added");
    }
    traverse(){
        if(this.head==null){
            console.log("Linked List is empty");
        }else{
            for(let temp=this.head;temp!=null;temp=temp.next){
                console.log("Data is "+temp.data);
                console.log("------------------------------");
            }
        }
    }
}

// Let's Start here
ll=new LinkedListDS();
ll.addNode(10);
ll.addNode(20);
ll.addNode(30);
ll.addNode(40);
ll.addNode(50);
ll.traverse();

File: 100-Days-Series/Day 48/LinkedListDS.py
class LinkedList:
    def __init__(self,data):
      self.data=data
      self.next=None

class LinkedListDS:
    def __init__(self):
      self.head=None

    def addNode(self,data):
        newNode=LinkedList(data)
        if self.head==None:
           self.head=newNode
           newNode.next=None
        else:
            temp=self.head
            last=self.head
            while temp!=None:
              last=temp
              temp=temp.next
            last.next=newNode
            newNode.next=None
        print('New Node Added')
    
    def traverse(self):
        if  self.head==None:
          print('Linked list is empty')
        else:
           temp=self.head
           while temp!=None:
              print('Data is ',temp.data)
              print('---------------------')
              temp=temp.next

# Let's Start here
ll=LinkedListDS()
ll.addNode(10)
ll.addNode(20)
ll.addNode(30)
ll.addNode(40)
ll.addNode(50)
ll.traverse()
    